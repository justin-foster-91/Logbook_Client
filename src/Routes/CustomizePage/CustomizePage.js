import React, { Component } from 'react';
import './CustomizePage.css';
import CustomizeParts from '../../Components/CustomizeParts/CustomizeParts';
import ShipApiService from '../../Services/ship-api-service';


class CustomizePage extends Component {
  state = {ship: undefined}

  componentDidMount() {
    console.log(this.props.ships);
    ShipApiService.getShip(this.props.match.params.id)
    .then(ship => {
      this.setState({
        ship: ship
      })
      console.log(ship);
    })

  }

  render() {
    return (
      <div>
        <h1>
          {"Hello Customize Page " + this.props.match.params.id}
        </h1>
        {this.state.ship ? <CustomizeParts ship={this.state.ship}/> : <div>Waiting for ship</div>}

      </div>
    );
  }
}

export default CustomizePage;

// const CustomizePage = (props) => {

//   return (
//     <div>

//     </div>
//   );
// }

// CustomizePage.defaultProps = {
//   history: {
//     push: () => {},
//   },
// }

// export default CustomizePage;