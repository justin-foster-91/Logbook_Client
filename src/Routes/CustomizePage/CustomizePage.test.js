import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CustomizePage from './CustomizePage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter><CustomizePage match={{params: {id: 1}}} /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<BrowserRouter><CustomizePage match={{params: {id: 1}}} /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});
