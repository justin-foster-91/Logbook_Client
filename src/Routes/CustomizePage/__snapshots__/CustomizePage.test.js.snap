// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <h1>
    Ship Customizer
  </h1>
  <div>
    Ship docking. Please wait.
  </div>
</div>
`;
